src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactory.java

```diff
@@ -1,41 +1,41 @@
 package io.github.oliviercailloux.y2018.apartments.apartment;
 
-import io.github.oliviercailloux.y2018.apartments.apartment.Apartment.Builder;
-import io.github.oliviercailloux.y2018.apartments.utils.JsonConvert;
-
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.MalformedURLException;
-import java.net.URL;
+import java.io.InvalidObjectException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
 
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import io.github.oliviercailloux.y2018.apartments.apartment.Apartment.Builder;
+import io.github.oliviercailloux.y2018.apartments.utils.JsonConvert;
+
 /**
  * A factory for creating Apartment objects.
  */
 public abstract class ApartmentFactory {
 
-	private static final int NB_MAX_RETRY = 5;
-
 	private static final Logger LOGGER = LoggerFactory.getLogger(ApartmentFactory.class);
 
 	private static Random rand = new Random();
 
 	/** The URL of the API address. */
-	private static final String URL_API_ADDRESS = "https://8n8iajahab.execute-api.us-east-1.amazonaws.com/default/RealRandomAdress";
+	private static final String URL_API_ADDRESS = "https://api-adresse.data.gouv.fr/reverse/";
 
 	/**
 	 * This function aims to generate a new apartment with random characteristics
 	 * and a real address.
 	 * 
-	 * @return An Apartment with random characteristics and a real address
+	 * @return An {@link Apartment} with random characteristics and a real address
```

Comment:
Un lien ici risque de distraire le lecteur, sans rien apporter, puisque c’est également lié via le type de retour, comme vous pouvez le vérifier. (OPT)

Reply:
Fix in a80a8f27fd1a55806ab72635368de4f7dac3f16d